### PR TITLE
Pricecaster V2-alpha

### PR DESCRIPTION
**Pull request by hernandp**
This PR contains the Pricecaster components (backend + contracts) that successfully:

* fetch VAAs from a spy host/guardian p2p network,
* construct a TX group to submits VAAs for validating through vaa-processor and vaa-verify TEAL code.
* publish data to a "pricekeeper" contract previously deployed.

Several enhancements such as documentation, logging system,  debugging facilities, and deployment tool are included.

### CODE REVIEW

**Comment by jumpsiegel** [comment]
No changes outside staged/algorand...
need to add governance/upgrade support before it can really go into production

**Review by jumpsiegel — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -10,4 +10,7 @@ venv
 .env
 bigtable-admin.json
 bigtable-writer.json
+DEPLOY-*
+VAA-VERIFY-*
```
Comment (leoluk) [inline .gitignore:14]: Do these have to be at the top level, or can we move them to staging/algorand/.gitignore?

File: staging/algorand/backend/fetcher/PythPriceFetcher.ts [hunk 2]
```diff
@@ -1,59 +1,62 @@
...
-    return this.strategy.getPrice()
-  }
-
-  private onPriceChange (price: PriceData) {
-    const pt: PriceTicker = new PriceTicker(price.priceComponent,
-      price.confidenceComponent, price.exponent, price.publishSlot)
-    this.strategy.put(pt)
-  }
-}
+// /**
```
Comment (leoluk) [inline staging/algorand/backend/fetcher/PythPriceFetcher.ts:1]: Just delete the file?

File: staging/algorand/backend/engine/PriceKeeperEngine.ts [hunk 3]
```diff
@@ -1,101 +1,101 @@
-/**
- * Pricecaster Service.
- *
- * Fetcher backend component.
- *
- * (c) 2021 Randlabs, Inc.
- */
+// /**
```
Comment (leoluk) [inline staging/algorand/backend/engine/PriceKeeperEngine.ts:1]: Just delete the file?

**Review by leoluk — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (hernandp) [reply .gitignore:14]: We can even remove those items, as they are deployment-only output files.

File: staging/algorand/backend/engine/PriceKeeperEngine.ts (hunk 3 above)
Comment (hernandp) [reply staging/algorand/backend/engine/PriceKeeperEngine.ts:1]: Agreed.

File: staging/algorand/backend/fetcher/PythPriceFetcher.ts (hunk 2 above)
Comment (hernandp) [reply staging/algorand/backend/fetcher/PythPriceFetcher.ts:1]: Agreed.

**Pull request merged**